b2sdk/transfer/emerge/executor.py

```diff
@@ -363,13 +368,26 @@ def _match_unfinished_file_if_possible(
         This is only possible if the application key being used allows ``listFiles`` access.
         """
         file_retention = file_retention or NO_RETENTION_FILE_SETTING
+        file_info_without_large_file_sha1 = {
+            k: v
+            for k, v in file_info.items() if k != LARGE_FILE_SHA1
+        }
         for file_ in self.services.large_file.list_unfinished_large_files(
             bucket_id, prefix=file_name
         ):
             if file_.file_name != file_name:
                 continue
             if file_.file_info != file_info:
-                continue
+                if (LARGE_FILE_SHA1 in file_.file_info) == (LARGE_FILE_SHA1 in file_info):
+                    # large_file_sha1 is present or missing in both file infos
```

Comment:
this is an excellent debug log statement, please use a logger. I tried to debug this code last week and it doesn't put anything remotely useful in logs.

While you are at it, please add log statements to all branches in this method so that we know what is going on.